### PR TITLE
(Windows) New table: default_environment

### PR DESCRIPTION
**Pull request by woodruffw**
Adds the `default_environment` table, which contains rows of default environment variables added to each process on initialization by Windows.

This table proxies and extrapolates its contents from the `registry` table.

### CODE REVIEW

File: tests/integration/tables/default_environment.cpp [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  }
+};
+
+TEST_F(defaultEnvironment, test_sanity) {
+  auto const data = execute_query("select * from default_environment");
+
+  ASSERT_GE(data.size(), 0ul);
+
+  ValidatatioMap row_map = {
+      {"variable", NormalType},
```
Comment (akindyakov) [inline tests/integration/tables/default_environment.cpp]: Could env variable be empty string? If not, can we check it here with `NonEmptyString` ?

Comment (woodruffw) [reply tests/integration/tables/default_environment.cpp]: I don't believe it can. I'll change it to `NonEmptyString` 👍

**Comment by woodruffw** [comment]
This has also been rebased and is good for another review (cc @alessandrogario / @theopolis / @Smjert).

**Comment by theopolis** [comment]
We've addressed the CI failure. If you don't mind rebasing onto master, this should fix the issue.

File: osquery/tables/system/windows/default_environment.cpp [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#include <osquery/core.h>
```
Comment (theopolis) [inline osquery/tables/system/windows/default_environment.cpp]: nitpick, include default license header

File: osquery/tables/system/windows/default_environment.cpp [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+namespace osquery {
+namespace tables {
+static std::string kEnvironmentKey =
+    "HKEY_LOCAL_MACHINE\\SYSTEM\\CurrentControlSet\\Control\\Session "
+    "Manager\\Environment";
+
+QueryData genDefaultEnvironment(QueryContext& context) {
+  QueryData results;
+  auto environment =
+      SQL::selectAllFrom("registry", "key", EQUALS, kEnvironmentKey);
```
Comment (theopolis) [inline osquery/tables/system/windows/default_environment.cpp]: Do you mind swapping this to use `queryKey` (from ./tables/system/windows/registry.h)?

File: osquery/tables/system/windows/default_environment.cpp [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+  for (const auto& env : environment) {
+    Row r;
+    r["variable"] = std::move(env.at("name"));
```
Comment (theopolis) [inline osquery/tables/system/windows/default_environment.cpp]: Are the `std::move`s relevant here? The const interator forces the `at` to be `const T&`?

File: tests/integration/tables/default_environment.cpp [hunk 5]
```diff
@@ -0,0 +1,40 @@
+
```
Comment (theopolis) [inline tests/integration/tables/default_environment.cpp]: nitpick, stray newline

File: osquery/tables/system/windows/default_environment.cpp [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+ */
+
+#include <osquery/core.h>
+#include <osquery/logger.h>
+#include <osquery/tables.h>
+#include <osquery/tables/system/windows/registry.h>
+
+namespace osquery {
+namespace tables {
+static std::string kEnvironmentKey =
```
Comment (zwass) [inline osquery/tables/system/windows/default_environment.cpp]: `const`

Comment (theopolis) [reply osquery/tables/system/windows/default_environment.cpp]: doh

**Pull request merged**